src/SAML2/Provider/HostedIdentityProviderProcessor.php

```diff
@@ -204,7 +219,8 @@ public function onAuthenticationSuccess(CoreAuthenticationEvent $event)
         }
 
         $user = $event->getAuthenticationToken()->getUser();
-        if($user instanceof UserInterface && $this->stateHandler->has()){
+        if ($this->stateHandler->get() !== null
+            && $user instanceof UserInterface && $this->stateHandler->has()) {
```

Comment:
This is equivalent, this `stateHandler->get() !== null` is the method `$stateHandler->has()`